services/gitlab/gitlab-pipeline-status.service.js

```diff
@@ -73,18 +78,65 @@ class GitlabPipelineStatus extends BaseSvgScrapingService {
     return renderBuildStatusBadge({ status })
   }
 
+  async fetch({ user, repo, branch, baseUrl }) {
+    const url = `${baseUrl}/${user}/${repo}/badges/${branch}/pipeline.svg`
+    const options = { headers: { Accept: 'image/svg+xml' } }
+    const errorMessages = {
+      401: 'repo not found',
+      404: 'repo not found',
+    }
+    try {
+      const { buffer } = await this._request({
+        url,
+        options,
+        errorMessages,
+      })
+      const data = {
+        status: super.constructor.valueFromSvgBadge(buffer),
+      }
+
+      return super.constructor._validate(data, badgeSchema)
+    } catch (e) {
+      if (e instanceof Inaccessible) {
+        try {
+          await this._request({
+            url,
+            options: {
+              ...options,
+              ...{ method: 'HEAD', followRedirect: false },
+            },
+            errorMessages,
+          })
+        } catch (err) {
+          const {
+            response: {
+              statusCode,
+              headers: { location },
+            },
+          } = err
+          if (
+            (statusCode === 301 || statusCode === 302) &&
+            location === `${baseUrl}/users/sign_in`
+          ) {
+            throw new NotFound({ prettyMessage: 'repo not found or private' })
+          }
+        }
```

Comment:
Could avoid the try/catch block wrapping around this HEAD request by using parent class internals like `_requestFetcher` but I'd prefer to leverage the util functions when possible, just requires the catch block here since `checkErrorResponse` will treat the 301/302 status codes as errors and throw an InvalidResponse